File: One-Time Pad (OTP) – Theoretically Unbreakable.cpp
#include <iostream>
#include <string>
#include <cstdlib>
#include <ctime>

using namespace std;

// Function to generate a random key of given length
string generate_random_key(int length) {
    string key;
    for (int i = 0; i < length; i++) {
        key += char(rand() % 256); // Generate random ASCII character
    }
    return key;
}

// Function to encrypt using One-Time Pad
string one_time_pad_encrypt(const string& plain_text, const string& key) {
    if (plain_text.length() != key.length()) {
        throw invalid_argument("Key length must match plaintext length");
    }
    
    string cipher_text;
    for (size_t i = 0; i < plain_text.length(); i++) {
        cipher_text += plain_text[i] ^ key[i]; // XOR operation
    }
    return cipher_text;
}

// Function to decrypt using One-Time Pad
string one_time_pad_decrypt(const string& cipher_text, const string& key) {
    return one_time_pad_encrypt(cipher_text, key); // Decryption is same as encryption
}

int main() {
    srand(time(0)); // Seed for randomness

    string plain_text = "Hello, Secure World!";
    string key = generate_random_key(plain_text.length());

    string encrypted_text = one_time_pad_encrypt(plain_text, key);
    string decrypted_text = one_time_pad_decrypt(encrypted_text, key);

    cout << "Plain Text: " << plain_text << endl;
    cout << "Generated Key: " << key << endl;
    cout << "Encrypted Text: ";
    for (char c : encrypted_text) cout << hex << (int)(unsigned char)c << " "; // Print in hex
    cout << endl;
    cout << "Decrypted Text: " << decrypted_text << endl;

    return 0;
}

File: One-Time Pad (OTP) – Theoretically Unbreakable.py
import base64
import random
import string

def one_time_pad_encrypt(plain_text, key):
    if len(key) < len(plain_text):
        raise ValueError("Key must be at least as long as the plaintext")
    cipher_text = ''.join(chr(ord(p) ^ ord(k)) for p, k in zip(plain_text, key))
    return base64.b64encode(cipher_text.encode()).decode()

def one_time_pad_decrypt(cipher_text, key):
    cipher_text = base64.b64decode(cipher_text.encode()).decode()
    return ''.join(chr(ord(c) ^ ord(k)) for c, k in zip(cipher_text, key))

def generate_random_key(length):
    return ''.join(random.choices(string.ascii_letters + string.digits, k=length))

# Example Usage
plain_text = "Hello, Secure World!"
key = generate_random_key(len(plain_text))

# One-Time Pad Encryption and Decryption
otp_encrypted = one_time_pad_encrypt(plain_text, key)
print("OTP Encrypted:", otp_encrypted)
print("OTP Decrypted:", one_time_pad_decrypt(otp_encrypted, key))
